addons/payment/tests/test_flows.py

```diff
@@ -88,39 +90,82 @@ def _test_flow(self, flow):
 
         return tx_sudo
 
+    @mute_logger('odoo.addons.payment.models.payment_transaction')
+    def _test_max_one_payment_request_from_portal(self, flow):
+
+        payment_option_id = self.create_token().id if flow == 'token' else self.acquirer.id
+        data = {
+            'amount': self.amount,
+            'currency_id': self.currency.id,
+            'partner_id': self.partner.id,
+            'access_token': None, flow: flow,
+            'payment_option_id': payment_option_id,
+            'reference_prefix': 'test',
+            'tokenization_requested': True,
+            'landing_route': 'Test',
+            'is_validation': False,
+            'invoice_id': self.invoice.id,
+            'flow': flow,
+        }
+
+        with unittest.mock.patch(
+                 'odoo.addons.payment.utils.check_access_token', return_value=True
+             ), \
+             unittest.mock.patch(
+                 'odoo.addons.payment.models.payment_transaction.PaymentTransaction.'
+                 '_send_payment_request'
+             ) as patched:
+
+            self.url_open(
```

Comment:
Might be worth a little comment to explain why it couldn't be done with the utils.
Also, did you check whether the utils could not be converted to this `url_open` usage so that you can directly check and patch `_send_payment_request` in the the `_test_flow` method?
I find it potentially overkill to add another pass through the `/payment/transaction` route in the 3 flows :thinking:.

Reply:
I choose to separate this test from the test_flow, because the test flow only check the values, without passing through the controller. So I felt it was a different thing that required a new function.

So, when I made it work with `url_open` and 'unittest.mock.patch', I was like: "It should have worked before, there literally was no reason for `portal_transaction` not working." I tested it, and indeed, it was working. I was the error when we spoke about it, and since I succeeded with both `url_open` (which has the favor of the ORM) and 'unittest.mock.patch', I didn't think twice about it again.  
So yes, I'll change `portal_transaction` anyway, since according to the ORM, `url_open` should be used instead of bypassing it to use `opener`. But there wasn't any problem with `portal_transaction` to begin with, just the way I used it... >.<